pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -2066,6 +2068,38 @@ func (gce *GCECloud) List(filter string) ([]string, error) {
 	return instances, nil
 }
 
+// GetAllZones returns all the zones in which nodes are running
+func (gce *GCECloud) GetAllZones() (sets.String, error) {
```

Comment:
Correct.  The GCE PD creation code is split between here and the volumes package; AWS has a cleaner separation of code.  So the zone is determine in the cloudprovider in AWS, and in the volumes code in GCE.

Note that this method is public, but is not part of the public interface.  i.e. it is public on GCECloud, not cloudprovider.Interface